src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private StorageDao storageDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+        storageDao = new StorageDaoImpl();
```

Comment:
Consider using dependency injection to pass the `StorageDao` instance instead of instantiating it directly. This will improve testability and flexibility.